feat: added custom classnames for event-limits tab

ibex088 opened the pull request:
## What does this PR do?

This PR introduces customizable class names (customClassnames) for the Event Type limits page in the project. This addition allows developers to style specific elements of the Event Type limits page

#### Below is a full list of available custom class names with default styling:

  ```json
{
    "eventLimitsTab": {
      "bufferAndNoticeSection": {
        "container": "border border-purple-200 bg-purple-100 rounded-lg",
        "beforeBufferSelect": {
          "container": "mb-4",
          "label": "text-purple-700",
          "select": "border border-purple-300 rounded-md text-purple-900 bg-white",
          "innerClassNames": {
            "input": "text-purple-900",
            "option": "text-purple-700 hover:bg-purple-200",
            "control": "border-purple-300",
            "singleValue": "text-purple-900",
            "valueContainer": "bg-purple-100",
            "menu": "border border-purple-200 rounded-md",
            "menuList": "bg-white"
          }
        },
        "afterBufferSelect": {
          "container": "mb-4",
          "label": "text-purple-700",
          "select": "border border-purple-300 rounded-md text-purple-900 bg-white",
          "innerClassNames": {
            "input": "text-purple-900",
            "option": "text-purple-700 hover:bg-purple-200",
            "control": "border-purple-300",
            "singleValue": "text-purple-900",
            "valueContainer": "bg-purple-100",
            "menu": "border border-purple-200 rounded-md",
            "menuList": "bg-white"
          }
        },
        "minimumNoticeInput": {
          "container": "mb-4",
          "label": "text-purple-700",
          "input": "border border-purple-300 rounded-md text-purple-900 bg-white",
          "select": "border border-purple-300 rounded-md text-purple-900 bg-purple-50",
          "innerClassNames": {
            "input": "text-purple-900",
            "option": "text-purple-700 hover:bg-purple-200",
            "control": "border-purple-300",
            "singleValue": "text-purple-900",
            "valueContainer": "bg-purple-100"
          }
        },
        "timeSlotIntervalSelect": {
          "container": "mb-4",
          "label": "text-purple-700",
          "select": "border border-purple-300 rounded-md text-purple-900 bg-white",
          "innerClassNames": {
            "input": "text-purple-900",
            "option": "text-purple-700 hover:bg-purple-200",
            "control": "border-purple-300",
            "singleValue": "text-purple-900",
            "valueContainer": "bg-purple-100"
          }
        }
      },
      "bookingFrequencyLimit": {
        "container": "border border-purple-200 bg-purple-100 rounded-lg",
        "label": "text-purple-700",
        "description": "text-purple-500",
        "children": "space-y-4",
        "intervalLimitContainer": "space-y-2",
        "intervalLimitItem": {
          "addLimitButton": "text-purple-500 hover:bg-purple-200 rounded-md",
          "limitText": "border border-purple-300 rounded-md text-purple-900 bg-white",
          "limitSelect": {
            "select": "border border-purple-300 rounded-md text-purple-900 bg-purple-50",
            "innerClassNames": {
              "input": "text-purple-900",
              "option": "text-purple-700 hover:bg-purple-200",
              "control": "border-purple-300",
              "singleValue": "text-purple-900",
              "valueContainer": "bg-purple-100"
            }
          },
          "container": "flex items-center space-x-2"
        }
      },
      "firstAvailableSlotOnly": {
        "container": "border border-purple-200 bg-purple-100 rounded-lg",
        "label": "text-purple-700",
        "description": "text-purple-500",
        "children": "space-y-4"
      },
      "totalDurationLimit": {
        "container": "border border-purple-200 bg-purple-100 rounded-lg",
        "label": "text-purple-700",
        "description": "text-purple-500",
        "children": "space-y-4",
        "intervalLimitItem": {
          "addLimitButton": "text-purple-500 hover:bg-purple-200 rounded-md",
          "limitText": "border border-purple-300 rounded-md text-purple-900 bg-white",
          "limitSelect": {
            "select": "border border-purple-300 rounded-md text-purple-900 bg-purple-50",
            "innerClassNames": {
              "input": "text-purple-900",
              "option": "text-purple-700 hover:bg-purple-200",
              "control": "border-purple-300",
              "singleValue": "text-purple-900",
              "valueContainer": "bg-purple-100"
            }
          },
          "container": "flex items-center space-x-2"
        }
      },
      "futureBookingLimit": {
        "container": "border border-purple-200 bg-purple-100 rounded-lg",
        "label": "text-purple-700",
        "description": "text-purple-500",
        "children": "space-y-4",
        "rollingLimit": {
          "container": "flex items-center space-x-2",
          "textField": "border border-purple-300 rounded-md text-purple-900 bg-white",
          "periodTypeSelect": {
            "select": "border border-purple-300 rounded-md text-purple-900 bg-purple-50",
            "innerClassNames": {
              "input": "text-purple-900",
              "option": "text-purple-700 hover:bg-purple-200",
              "control": "border-purple-300",
              "singleValue": "text-purple-900",
              "valueContainer": "bg-purple-100"
            }
          }
        },
        "rangeLimit": {
          "wrapper": "flex items-center space-x-2",
          "datePickerWraper": "border border-purple-300 rounded-md bg-white",
          "datePicker": "text-purple-900 bg-white"
        }
      },
      "offsetStartTimes": {
        "container": "border border-purple-200 bg-purple-100 rounded-lg",
        "label": "text-purple-700",
        "description": "text-purple-500",
        "children": "space-y-4",
        "offsetInput": {
          "container": "flex items-center space-x-2",
          "label": "text-purple-700",
          "input": "border border-purple-300 rounded-md text-purple-900 bg-white",
          "addOn": "text-purple-500"
        }
      }
    }
}
```

#### Screenshot of Changes

![Screenshot 2024-11-08 at 4 31 50 PM](https://github.com/user-attachments/assets/a6e1acc5-e495-4ef7-bf96-30213359a1ef)




## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Write details that help to start the tests -->

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

<!-- Remove bullet points below that don't apply to you -->

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

ibex088 committed:
added custom classnames for event limits tab

ibex088 committed:
added classNames to common types

ibex088 committed:
updated classnames to use the common exported types

ibex088 committed:
undo

ibex088 committed:
Update EventTypePlatformWrapper.tsx

ibex088 committed:
Merge branch 'main' into event-limits-tab-classnames

ibex088 committed:
Merge branch 'main' into event-limits-tab-classnames

ibex088 committed:
better clasnames naming

ibex088 committed:
added example classnames for demo

supalarry commented on packages/features/eventtypes/components/tabs/limits/EventLimitsTab.tsx:
```diff
@@ -23,7 +24,40 @@ import { Button, DateRangePicker, InputField, Label, Select, SettingsToggle, Tex
+export type EventLimitsTabCustomClassNames = {
+  bufferAndNoticeSection?: {
+    container?: string;
+    beforeBufferSelect?: SelectClassNames;
```
Question - what does `innerClassNames` mean within `SelectClassNames` ? These are classnames already existing in the code?

I am asking because if we have before buffer select:
```"beforeBufferSelect": {
        "container": "mb-4",
        "label": "text-purple-700",
        "select": "border border-purple-300 rounded-md text-purple-900 bg-white",
        "innerClassNames": {
          "input": "text-purple-900",
          "option": "text-purple-700 hover:bg-purple-200",
          "control": "border-purple-300",
          "singleValue": "text-purple-900",
          "valueContainer": "bg-purple-100",
          "menu": "border border-purple-200 rounded-md",
          "menuList": "bg-white"
        }
      },
```

we have `select` key and then `innerClassNames`. I wonder if we should rename `select` to `container` and `innerClassNames` to `select` and then move `container` within the `select` so we would have:
```
"beforeBufferSelect": {
        "container": "mb-4",
        "label": "text-purple-700",
        "select": {
          "container": "border border-purple-300 rounded-md text-purple-900 bg-white",
          "input": "text-purple-900",
          "option": "text-purple-700 hover:bg-purple-200",
          "control": "border-purple-300",
          "singleValue": "text-purple-900",
          "valueContainer": "bg-purple-100",
          "menu": "border border-purple-200 rounded-md",
          "menuList": "bg-white"
        }
      },
```

supalarry replied:
and the same for others

SomayChauhan replied:
https://github.com/calcom/cal.com/blob/48c0676f411e97cbbe973607004af2e2ac8ff114/packages/ui/components/form/select/Select.tsx#L27
yes -- these are classnames we already have in our ui/select component

ibex088 committed:
fix: better calssnames

bookingFrequencyLimitSection-> bookingFrequencyLimit
totalDurationLimitSection-> totalDurationLimit
futureBookingLimitSection-> futureBookingLimit
offsetStartTimesSection-> offsetStartTimes

ibex088 committed:
suffix->addOn

SomayChauhan commented:
> we have suffix there, but in other places we have addOn - is there a reason for that?

this was a mistake, fixed in https://github.com/calcom/cal.com/pull/17478/commits/51696e988ef64756efdba7ea105e848a4fc6d998

ibex088 committed:
Merge branch 'main' into event-limits-tab-classnames

ibex088 committed:
Update types.ts

ibex088 committed:
Merge branch 'main' into event-limits-tab-classnames

ibex088 committed:
Merge branch 'main' into event-limits-tab-classnames

ibex088 committed:
Merge branch 'main' into event-limits-tab-classnames
